modules/gdscript/language_server/godot_lsp.h

```diff
@@ -1972,15 +2016,120 @@ static String marked_documentation(const String &p_bbcode) {
+					String text = line.substr(url_end + 1, close_start - url_end - 1);
+					String replacement = "[" + text + "](" + url + ")";
+					line = line.substr(0, pos) + replacement + line.substr(close_start + URL_CLOSE_TAG_LENGTH);
+					pos += replacement.length();
+				} else {
+					break;
+				}
+			}
+
+			// This is for the case [url]$url[/url]
+			pos = 0;
+			while ((pos = line.find("[url]", pos)) != -1) {
+				int close_pos = line.find("[/url]", pos);
+				if (close_pos != -1) {
+					String url = line.substr(pos + URL_OPEN_TAG_LENGTH, close_pos - pos - URL_OPEN_TAG_LENGTH);
+					String replacement = "[" + url + "](" + url + ")";
+					line = line.substr(0, pos) + replacement + line.substr(close_pos + URL_CLOSE_TAG_LENGTH);
+					pos += replacement.length();
+				} else {
+					break;
+				}
+			}
+
+			// Replace the various link types with inline code. For example, [class MyNode] becomes `MyNode`
+			// Uses a while loop because there can occasionally be multiple links of the same type in a single line
+			const Vector<String> link_start_patterns = {
+				"[class ", "[method ", "[member ", "[signal ", "[enum ", "[constant ",
+				"[annotation ", "[constructor ", "[operator ", "[theme_item ", "[param "
+			};
+			for (const String &pattern : link_start_patterns) {
+				int pattern_pos = 0;
+				while ((pattern_pos = line.find(pattern, pattern_pos)) != -1) {
+					int end_pos = line.find("]", pattern_pos);
+					if (end_pos != -1) {
+						String content = line.substr(pattern_pos + pattern.length(), end_pos - pattern_pos - pattern.length());
+						String replacement = "`" + content + "`";
+						line = line.substr(0, pattern_pos) + replacement + line.substr(end_pos + 1);
+						pattern_pos += replacement.length();
+					} else {
+						break;
+					}
+				}
+			}
+
+			// Remove tags with attributes like [color=red] as they don't have a
+			// direct markdown equivalent supported by external tools.
+			const String attribute_tags[] = {
+				"color", "font", "img"
+			};
+			for (const String &tag_name : attribute_tags) {
+				int tag_pos = 0;
+				while ((tag_pos = line.find("[" + tag_name + "=", tag_pos)) != -1) {
+					int end_pos = line.find("]", tag_pos);
+					if (end_pos != -1) {
+						line = line.substr(0, tag_pos) + line.substr(end_pos + 1);
+					} else {
+						break;
+					}
+				}
+				line = line.replace("[/" + tag_name + "]", "");
+			}
+
+			// Convert remaining simple bracketed class names to backticks and literal brackets.
+			// This handles cases like [Node2D], [Sprite2D], etc. and [lb] and [rb].
+			pos = 0;
+			while ((pos = line.find("[", pos)) != -1) {
+				// Replace the special cases for [lb] and [rb] first and walk
+				// past them to avoid conflicts with class names.
+				const bool is_within_bounds = pos + 4 <= line.length();
+				if (is_within_bounds && line.substr(pos, 4) == "[lb]") {
+					line = line.substr(0, pos) + "[" + line.substr(pos + 4);
```

Comment:
```suggestion
					line = line.substr(0, pos) + "\[" + line.substr(pos + 4);
```

At the moment we don't handle escaping of special markdown characters anywhere. But at least for [ and ] it would make sense to escape them (the markdown way) since they are special characters in markdown as well.